Automattic/wp-calypso
# Add contextual trigger for Backup add-on products

#### Proposed Changes
* Add Backup Add-on to the calypso-package
* Update the upsell contextual trigger to show storage and price

#### Testing Instructions

* Create a JN site with Jetpack Backup 10GB product (or) Jetpack Security 10GB
*  click on Jetpack Cloud live link below
    * Goto `/backup`
* or boot up this PR 
    * Run `git fetch && git checkout add/backup-addon`
    * Run `yarn start-jetpack-cloud`
    * Goto http://jetpack.cloud.localhost:3000/backup
* Confirm that upsell contextual trigger is shown below the storage usage bar accordingly (refer to the screenshot for different scenarios and the respective message shown for the same
* Make sure 10GB Add-on is suggested for 10GB base plan 
* Make sure that clicking on the trigger takes you to the checkout screen (currently checkout page will throw error since we are yet to add the product to the store)
* Repeat the same for 1TB base plan (i.e Jetpack Backup 1TB or Jetpack Security 1TB) and make sure 1TB add-on is suggested 

**TIP:** You can simulate different storage usages with Redux DevTools in your browser. Dispatch the following action, substituting `<your_site_id>` with your site's numeric ID and `<bytes>` with a number of bytes used:

(For reference: 5TB = `5000000000`; 9GB = `9000000000`; 11GB = `11000000000`)

```js
{
  type: 'REWIND_SIZE_SET',
  siteId: <your_site_id>,
  size: {
    bytesUsed: <bytes>
  }
}
```
#### Screenshot
##### Jetpack Backup 10GB ( with Usage level Normal )
<img width="742" alt="Screenshot 2022-10-24 at 6 28 50 PM" src="https://user-images.githubusercontent.com/2027003/197536207-f585c2b4-4c50-4484-b028-356c608e5c7f.png">


##### Jetpack Backup 10GB ( with Usage level Warning )
<img width="742" alt="Screenshot 2022-10-24 at 6 29 29 PM" src="https://user-images.githubusercontent.com/2027003/197536870-a2b37b79-e2a5-40ad-a850-c478a88751d5.png">

##### Jetpack Backup 10GB ( with Usage level Critical )
<img width="739" alt="Screenshot 2022-10-24 at 6 56 24 PM" src="https://user-images.githubusercontent.com/2027003/197537082-b06bdfdd-f43b-4d6b-b85e-45bdc55881f0.png">

#### Jetpack Backup 10GB ( with Usage level Full )
<img width="741" alt="Screenshot 2022-10-24 at 6 29 54 PM" src="https://user-images.githubusercontent.com/2027003/197537201-4683934b-c463-48ae-b3ff-00aa4ec9ac98.png">



#### Pre-merge Checklist






Related to #
1203228880371291-as-1203228880371313/f
